agent/core/src/main/java/org/jolokia/restrictor/RestrictorFactory.java

```diff
@@ -35,6 +40,27 @@
  */
 public final class RestrictorFactory {
 
+    private static final Pattern URL_PATTERN = Pattern.compile("(http[s]?://)((.+):(.+)@)?(.+)");
+
+    /* default */ static class URLConnectionResult {
+
+        public final URLConnection urlConnection;
+
+        public final String url;
+
+        // URLConnection isn't queryable for "security headers";
+        // make this available for unit testing
+        public final String basicAuthHeaderValue;
+
+        private URLConnectionResult(URLConnection urlConnection, String url,
+                String basicAuthHeaderValue) {
+            this.urlConnection = Objects.requireNonNull(urlConnection);
```

Comment:
Yikes. Made the same mistake with Base64 and switched to Base64Util virtually just before commit. Whyever maven is not nagging as source and target source level are set to 1.5. Eclipse with m2e doesn't nag either.

Ok to skip the check alltogether then?